src/types/commands.ts

```diff
@@ -498,42 +499,44 @@ export interface CreateFigureCommand extends SheetDependentCommand {
   figure: Figure;
 }
 
-export interface UpdateFigureCommand extends Partial<Figure>, SheetDependentCommand {
+export interface UpdateFigureCommand extends Omit<Partial<Figure>, "id">, SheetDependentCommand {
   type: "UPDATE_FIGURE";
-  id: UID;
+  figureId: UID;
 }
 
 export interface DeleteFigureCommand extends SheetDependentCommand {
   type: "DELETE_FIGURE";
-  id: UID;
+  figureId: UID;
+}
+
+interface BaseFigureCommand extends SheetDependentCommand {
+  figureId: UID;
+  anchor?: Position;
```

Comment:
I don't think `anchor` is transformed if someone were to add a row at the same time you add a figure. Would work w/o needing `ot_specific` stuff if you extends `PositionDependentCommand` and remove the anchor.